### DISCUSSION TITLE
PyTorch fundamentals exercise

### DISCUSSION THREAD

**Discussion by NEELANATARAJAN** [mrdbourke/pytorch-deep-learning#1152] [Q&A] [answered]
Question 4: Set the random seed to 0 and do 2 & 3 over again.
The output should be:

(tensor([[1.8542],
         [1.9611],
         [2.2884],
         [3.0481],
         [1.7067],
         [2.5290],
         [1.7989]]), torch.Size([7, 1]))

If we initiate the seed using the manual_seed command and create two tensors and matmul with 2nd tensor transposed we are getting the desired result

<img width="596" alt="image" src="https://github.com/user-attachments/assets/92660fb5-27f2-4142-b7bb-2b29e0bdc167" />

But as mentioned in the course if the manual_seed command is called before creating the input tensors separately, the results vary 

<img width="644" alt="image" src="https://github.com/user-attachments/assets/01692a8e-c44f-4327-8f3a-5bc0c65c862f" />

Can someone explain which is the right way to handle reproducibility? Should seed command be initiated before every tensor creation or only once before all the commands?

**Comment by NEELANATARAJAN** [comment]
Author has called twice separately in two different cells in the lecture. I did the same and the results do not match hence the query.

**Comment by pem131** [comment, marked as answer]
Hello,

A good practice to obtain reproducibility is to set a seed at the beginning of your code (top of your notebook cell) and stick with the same seed number in your experiments.

Best wishes.
